src/ast/typescript/typeScriptAstUtils.kt

```diff
@@ -222,6 +224,13 @@ fun TS.UnionTypeNode.toKotlinTypeName(typeMapper: ObjectTypeToKotlinTypeMapper):
     return DYNAMIC + commentWithExpectedType
 }
 
+fun TS.IntersectionTypeNode.toKotlinTypeName(typeMapper: ObjectTypeToKotlinTypeMapper): String {
+    val kotlinTypeNames = types.arr.map { it.toKotlinTypeName(typeMapper) }
+    val commentWithExpectedType = kotlinTypeNames.joinToString(" & ", prefix = " /* ", postfix = " */")
+    // just take the first one for now since Kotlin doesn't support intersection types.
+    return kotlinTypeNames[0] + commentWithExpectedType
```

Comment:
right now for functions we can emulate intersection types with [upper-bounds](https://kotlinlang.org/docs/reference/generics.html#upper-bounds), like:
```kotlin
fun <T> foo(): T where T : Number, T : Boo {

}
```